Makefile

```diff
@@ -46,16 +49,47 @@ develop: publish-build
 	docker-compose logs -f
 
 # Clean
+clean: TAG := latest
 clean:
 	docker-compose stop
 	docker-compose down --remove-orphans --volumes --rmi all
 	docker-compose rm -svf
-	docker rmi --force openwisp/openwisp-base:latest \
-				openwisp/openwisp-base:intermedia-system \
-				openwisp/openwisp-base:intermedia-python \
+	docker rmi --force $(DOCKER_USERNAME)/openwisp-base:$(TAG) \
+				$(DOCKER_USERNAME)/openwisp-base:intermedia-system \
+				$(DOCKER_USERNAME)/openwisp-base:intermedia-python \
+				`docker images -f "dangling=true" -q` || true
+
+release-clean: clean
+	docker-compose rm -svf
+	docker rmi --force $(DOCKER_USERNAME)/openwisp-base:$(TAG) \
+				$(DOCKER_USERNAME)/openwisp-base:intermedia-system \
+				$(DOCKER_USERNAME)/openwisp-base:intermedia-python \
 				`docker images -f "dangling=true" -q` || true
 
 # Publish
+publish: TAG := latest
 publish: publish-build develop-runtests
-	docker push openwisp/openwisp-base:latest
+	docker login -u $(DOCKER_USERNAME)
```

Comment:
I do agree, but sometimes user just forgot if they were logged in or not. It's kind of silly if the makefile stops at the docker push and you have to login first, then call make again

Reply:
But yeah, I can remove it if you want :wink:

Reply:
I am not particular about it, I just think 12 factor asks us to keep management / one-off commands seperate.

I'll defer this to @2stacks or @nemesisdesign

Reply:
I would expect the user to be logged in already.  I can't recall the last time I had to log in to Docker or Github.  It might be good that the execution stops as it would indicate that there's an issue with cached credentials.